File: python_stack/python_fundamentals/bike/bike.py
class Bike:
    def __init__(self,max_speed,miles, price):
        self.max_speed=max_speed
        self.miles=miles
        self.price=price
    def ride(self):
        self.miles=self.miles+10
        return self
    def reverse(self):
        self.miles=self.miles-5
        return self
    def displayinfo(self):
        print('miles: ',self.miles,', max speed: ',self.max_speed,', price: ',self.price)
        return self
bike1=Bike('30mph',10,'$250')
bike2=Bike('25mph',400,'$100')
bike3=Bike('15mph',20,'$275')
bike4=Bike('45mph',1200,'$1000')
bike1.ride().ride().ride().reverse().displayinfo()
bike2.ride().ride().reverse().reverse().displayinfo()
bike3.reverse().reverse().reverse().displayinfo()

File: MEAN/bike.ts
class Bike{
    constructor(public maxspeed: number,public price:number, public miles:number) {
        this.maxspeed = maxspeed;
        this.miles = miles;
        this.price = price;
    }
    displayinfo() {
        console.log(this.maxspeed, this.price, this.miles);
        return this;
    }
    ride() {
        this.miles += 10;
        return this;
    }
    reverse() {
        this.miles -= 6;
        return this
    }
}
var bike1 = new Bike(5, 5, 5);
var bike2 = new Bike(5, 5, 5);
var bike3 = new Bike(5, 5, 5);
bike1.ride().ride().ride().reverse().displayinfo();
bike2.ride().ride().reverse().reverse().displayinfo();
bike3.reverse().reverse().reverse().reverse().displayinfo();

